src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -2002,26 +2002,29 @@ void VectorBuilder::multiReduction(SmallVectorImpl<Value> &inputVecArray,
   }
 }
 
+// New style.
 /*static*/ int64_t VectorBuilder::computeSuitableUnrollFactor(
     MemRefType memRefType, int64_t collapsedInnermostLoops, GenOpsMix genOps,
-    int64_t &simdLoopStaticTripCount) {
-
-  simdLoopStaticTripCount = 0; // Initially assume no SIMD.
+    bool canOverCompute, int64_t &simdLoopStaticTripCount, bool &simdOnly) {
```

Comment:
It is a little weird to put this function in DialectBuilder. How about a simd-related utils in onnx to krnl?

Reply:
Thanks for the nudge, I totally agree. Moved the functions to the `ONNXToKrnlCommon` files